pkg/deckgen/generate.go

```diff
@@ -140,6 +140,9 @@ func ToDeckContent(
 		for _, v := range c.Oauth2Creds {
 			consumer.Oauth2Creds = append(consumer.Oauth2Creds, &v.Oauth2Credential)
 		}
+		for _, v := range c.MTLSAuths {
+			consumer.MTLSAuths = append(consumer.MTLSAuths, &v.MTLSAuth)
```

Comment:
Codecov is complaining that we didn't add tests for this new loop, how feasible is it to do so?

Reply:
We don't currently test `ToDeckContent()` at all. We should be able to test most of it, though the plugin bits require a Kong client. Building boilerplate state objects may be a bit annoying and verbose, similar to the parser, but it's doable.

Reply:
What do you feel is the rough amount of time effort? Are you feeling inclined to add this now? I'm open to making a follow up issue, but the thing with tests is often if you don't just do them and you backlog them, they may never get done :thinking:

Reply:
Unsure of effort given the lack of any prior art here. The monolithic "take all the Kong objects and copy them around" functions tend to be more annoying to unit test. We do have integration tests that handle at least one type of credential, and I wouldn't expect a failure here to be specific to a single credential type--`ToDeckContent()` doesn't do anything of interest with them, only copies like go-kong structs out of KIC structs into decK structs.

I'd say backlog this to test `ToDeckContent()` in general given that the mTLS change to it is more of the same pattern we use for other credential types, with an eye towards possibly refactoring it into smaller pieces so that we can test it without testing ALL OF THE OBJECTS.

Reply:
I'm OK with that course of action :+1: